pylint/checkers/classes/class_checker.py

```diff
@@ -1356,6 +1362,33 @@ def _check_slots(self, node):
                     self._check_slots_elt(elt, node)
                 except astroid.InferenceError:
                     continue
+            self._check_redefined_slots(node, slots, values)
+
+    def _check_redefined_slots(
+        self,
+        node: nodes.ClassDef,
+        slots_node: nodes.NodeNG,
+        slots_list: List[nodes.NodeNG],
+    ) -> None:
+        """Check if `node` redefines a slot which is defined in an ancestor class"""
+        slots_names = [getattr(slot, "value", None) for slot in slots_list]
```

Comment:
Using `getattr` here instead of `slot.value` as a test scenario fails which uses `deque.__name__` as a slot.